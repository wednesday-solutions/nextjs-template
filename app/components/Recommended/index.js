/**
 *
 * Recommended
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import { ClickableTags } from '../styled/repos';
import { Box } from '@mui/material';

const Recommended = (props) => {
  const { recommendations } = props;
  const router = useRouter();
  return (
    <Box sx={{ display: 'flex', gap: '10px' }} data-testid="recommended">
      {recommendations.map(({ id, name }) => (
        <div key={id} onClick={() => router.push(`/info/${name}`)}>
          <ClickableTags label={name} size="small" clickable />
        </div>
      ))}
    </Box>
  );
};

Recommended.propTypes = {
  recommendations: PropTypes.arrayOf(
    PropTypes.shape({ id: PropTypes.number.isRequired, name: PropTypes.string.isRequired })
  )
};

export default Recommended;
