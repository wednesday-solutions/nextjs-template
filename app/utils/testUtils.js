import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from '../configureStore';
import colors from '@themes/colors';
import { translationMessages, DEFAULT_LOCALE } from '../i18n';

export const getComponentStyles = (Component, props = {}) => {
  renderProvider(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);

  return window.getComputedStyle(componentRoots[0])._values;
};

export const renderProvider = (children) => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={colors}>{children}</ThemeProvider>
    </IntlProvider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const renderStoreProvider = (children) => {
  const store = configureStore({});
  return render(
    <Provider store={store}>
      <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
        <ThemeProvider theme={colors}>{children}</ThemeProvider>
      </IntlProvider>
    </Provider>
  );
};
