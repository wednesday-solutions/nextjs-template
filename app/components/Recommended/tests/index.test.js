/**
 *
 * Tests for Recommended
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom'
import { renderProvider } from '@utils/testUtils'
import Recommended from '../index'

describe('<Recommended />', () => {
  const recommendations = [
    {
      name: 'test repo name',
      id: 1
    }
  ]
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(
      <Recommended {...{ recommendations }} />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 Recommended component', () => {
    const { getAllByTestId } = renderProvider(
      <Recommended {...{ recommendations }} />
    )
    expect(getAllByTestId('recommended').length).toBe(1)
  })
})
