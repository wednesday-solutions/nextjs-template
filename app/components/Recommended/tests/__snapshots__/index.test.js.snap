// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recommended /> should render and match the snapshot 1`] = `
.emotion-2 {
  cursor: pointer;
}

.emotion-2:hover {
  border: 1px solid #006ED6;
}

<body>
  <div>
    <div
      class="ant-row emotion-0"
      data-testid="recommended"
    >
      <div
        class="ant-col emotion-0"
      >
        <span
          class="ant-tag emotion-2 emotion-0"
        >
          test repo name
        </span>
      </div>
    </div>
  </div>
</body>
`;
